WordPress/gutenberg
# Spacer: Show tooltip with height value on resize

## Description

![Screen Capture on 2020-06-10 at 16-38-00](https://user-images.githubusercontent.com/2322354/84317813-6d190d80-ab3b-11ea-8f16-5bc29e227234.gif)

This update enhances the **Spacer** block by rendering a tooltip element above the cursor when the user engages in a "drag to resize" interaction.

#### ✨ New `<ResizeTooltip />` component

The tooltip comes from a new `<ResizeTooltip />` component. This component is a general resize-aware component that renders a tooltip on size changes. It isn't bound to `<ResizableBox />` (which is what Spacer uses), but it can be integrated with it.

As an aside... This component has 2 different positions:

* `bottom`
* `corner` (default)

##### 📄 Corner

![Screen Capture on 2020-06-10 at 16-06-28](https://user-images.githubusercontent.com/2322354/84318842-0eed2a00-ab3d-11ea-988a-af93697a9795.gif)

The Tooltip will render on the top/right of the parent element.

##### ⬇️ Bottom

<img width="569" alt="Screen Shot 2020-06-11 at 10 03 13 AM" src="https://user-images.githubusercontent.com/2322354/84394864-7a7dd880-abcb-11ea-921b-49b92dcee287.png">

The Tooltip will render on the bottom/center of the parent element.

#### 🎨 Design

As far as the Tooltip design goes... I went with a UI that was in the spirit of "G2". Something minimal, sporting the all black/white look.

## How has this been tested?

* Tested in local Storybook
* Tested in local Gutenberg

To test...

* Run `npm install`
* Run `npm run dev`
* Add a Spacer block
* Drag to resize. You should see the Tooltip.
* Drag the RangeSlider (sidebar) to resize the Spacer. You should NOT see the Tooltip.


## Types of changes

* Add `react-resize-aware` as dependency in `@wordpress/components` (it's also used in `@wrodpress/compose`)
* Add new `<ResizeTooltip />` component
* Exports `ResizeTooltip` from `@wordpress/components`
* Integrate `<ResizeTooltip />` into Spacer block

## Checklist:

The idea for this PR originally came from:
https://github.com/WordPress/gutenberg/issues/22956
